### PR TITLE
Parent child relation component

### PR DESCRIPTION
**Pull request by kpsherva**
First of the series of PRs to re-model relations component.
Styling will be worked on in the meantime, I would like to split this on smaller chunks. Please share your ideas.
Screenshot 2020-01-23 at 09 17 29

### CODE REVIEW

**Review by equadon — COMMENTED** [review]
This is a great improvement! It looks really nice and gives a great overview of the new relation.

I still have some reservations about the usage of _is volume X of_ since it's not changed yet and also how similar the green label looks to the confirm button. But we'll see what the feedback says later.

File: ui/src/pages/backoffice/components/buttons/ViewDetailsButtons/DocumentViewDetailsLink.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        {...this.props}
+      >
+        {this.props.children}
+      </Link>
+    );
+  }
+}
+
+DocumentViewDetailsLink.propTypes = {
+  documentPid: PropTypes.string.isRequired,
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply ui/src/pages/backoffice/components/buttons/ViewDetailsButtons/DocumentViewDetailsLink.js]: I would prefer to be explicit with naming props (and everything, naming police)
on the frontsite we usually don't have links under plain text, so I left this component in the backoffice only

File: ui/src/pages/backoffice/components/buttons/ViewDetailsButtons/DocumentViewDetailsLink.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { BackOfficeRoutes } from '@routes/urls';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+
+export class DocumentViewDetailsLink extends Component {
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply ui/src/pages/backoffice/components/buttons/ViewDetailsButtons/DocumentViewDetailsLink.js]: you are right, I will remove view from the name

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationRemover.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    return (
+      <Modal
+        trigger={trigger || <Button>{buttonContent}</Button>}
+        onClose={this.handleClose}
+        onOpen={this.handleOpen}
+        open={this.state.modalOpen}
+      >
+        <Modal.Header>{header && 'Confirm removal'}</Modal.Header>
+        <Modal.Content>
+          Are you sure you want to destroy this relation?
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationRemover.js]: 👍

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipartModal.js [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+  }
+
+  toggle = () => this.setState({ visible: !this.state.visible });
+
+  onSave = () => {
+    this.setState({ isLoading: true });
+    const { selection, volume } = this.state;
+
+    console.log(selection, 'parent');
+    console.log(this.props.documentDetails, 'child');
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipartModal.js]: 👀

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipart.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    };
+  }
+
+  onPageChange = page => {
+    this.setState({ activePage: page });
+  };
+
+  viewDetails = ({ row }) => {
+    return (
+      <SeriesViewDetailsLink seriesPid={row.pid} data-test={row.pid}>
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipart.js]: true

File: ui/src/components/Document/DocumentTitle.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export const DocumentTitle = ({ document }) => {
+  return `${document.metadata.title} (ed. ${document.metadata.edition}, ${document.metadata.publication_year})`;
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply ui/src/components/Document/DocumentTitle.js]: 👍

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationCard/RelationCard.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+            size="tiny"
+            onError={e => (e.target.style.display = 'none')}
+          />
+        )}
+        <Card.Content>
+          <Card.Header>{data.metadata.title}</Card.Header>
+          <Card.Meta>
+            <DocumentAuthors metadata={data.metadata} />
+            <div>
+              Ed. {data.metadata.edition}{' '}
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationCard/RelationCard.js]: 👍

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipartModal.js [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+
+    const newRelation = parentChildRelationPayload(
+      this.props.relationType,
+      { volume: volume },
+      selection,
+      this.props.documentDetails
+    );
+    console.log(newRelation, 'Payload');
```
Comment (topless) [inline ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipartModal.js]: Leftovers here too I guess

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipart.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (topless) [reply ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipart.js]: There are more occurrences of this in the PR

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipartModal.js [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+              volume={this.state.volume}
+            />
+          </Container>
+        </Modal.Content>
+        <Modal.Actions>
+          <Button onClick={() => this.toggle()}>Cancel action</Button>
+          <Button
+            positive
+            loading={this.state.isLoading}
+            disabled={this.state.isLoading}
```
Comment (topless) [inline ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipartModal.js:122]: It looks to me that this button, is either `loading` or `disabled`, so when is it clickable?

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationRemover.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+import { Button, Modal } from 'semantic-ui-react';
+
+export default class RelationRemover extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { modalOpen: false };
+  }
+
+  handleClose = () => this.setState({ modalOpen: false });
+  handleOpen = () => this.setState({ modalOpen: true });
```
Comment (topless) [inline ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationRemover.js:12]: Redundant 
```javascript
handleToggle = () => this.setState({ modalOpen: !this.state.modalOpen });
```

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/__tests__/DocumentRelations.test.js [hunk 11]
```diff
@@ -77,41 +96,41 @@ describe('Document relations tests', () => {
...
-      .simulate('click');
-    expect(mockViewDetails).toHaveBeenCalled();
-  });
+    // component
+    //   .find('TableRow')
+    //   .filterWhere(element => element.prop('data-test') === id)
+    //   .find('Button')
+    //   .simulate('click');
+    // expect(mockViewDetails).toHaveBeenCalled();
+  // });
```
Comment (topless) [inline ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/__tests__/DocumentRelations.test.js]: :) lets do something with it!

File: ui/src/pages/backoffice/components/NoResultsMessage/NoResultsMessage.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import { Icon, Message } from 'semantic-ui-react';
+
+export default class NoResultsMessage extends Component {
```
Comment (topless) [inline ui/src/pages/backoffice/components/NoResultsMessage/NoResultsMessage.js]: We use the term `EmptyResults` elsewhere, we should align

File: ui/src/pages/backoffice/components/buttons/ViewDetailsButtons/DocumentViewDetailsLink.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (topless) [reply ui/src/pages/backoffice/components/buttons/ViewDetailsButtons/DocumentViewDetailsLink.js]: On another small comment, the parent folder that holds all the components `ViewDetailsButtons` should be `DetailLinks` probably to reflect to the actual components. I am not a philologist but I think the plural should go only on the second part of the word.

**Review by topless — APPROVED** [review]
Splendid, a lot of complicated work went in here! 

On a side note `Prettier` can format also style files, format all the `*.overrides` files!!

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipartModal.js (hunk 9 above)
Comment (kpsherva) [reply ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipartModal.js:122]: when it is not loading

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/DocumentSiblings.js [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+          key: 'languages',
+          content: (
+            <>
+              Languages <Label>{languages.length}</Label>{' '}
+            </>
+          ),
+        },
+        render: () => <RelationMultipart />,
+      },
+      // {
```
Comment (topless) [inline ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/DocumentSiblings.js:25]: Is this used?

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipartModal.js (hunk 9 above)
Comment (topless) [reply ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationMultipartMonograph/RelationMultipartModal.js:122]: My bad I read it like it was negated

File: ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/DocumentSiblings.js (hunk 13 above)
Comment (kpsherva) [reply ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/DocumentSiblings.js:25]: no, but I want to make this change in steps, don't worry I will clean everything in the last PR

**Review by ntarocco — APPROVED** [review]

**Pull request merged**